WordPress/src/main/java/org/wordpress/android/ui/plans/PlanUpdateService.java

```diff
@@ -0,0 +1,173 @@
+package org.wordpress.android.ui.plans;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+import android.support.annotation.NonNull;
+
+import com.android.volley.VolleyError;
+import com.wordpress.rest.RestRequest;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.ui.plans.models.SitePlan;
+import org.wordpress.android.ui.prefs.AppPrefs;
+import org.wordpress.android.util.AppLog;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.greenrobot.event.EventBus;
+
+/**
+ * service which updates both global plans and available plans for a specific site
+ */
+
+public class PlanUpdateService extends Service {
+
+    private static final String ARG_LOCAL_BLOG_ID = "local_blog_id";
+    private int mLocalBlogId;
+
+    public static void startService(Context context, int localTableBlogId) {
+        Intent intent = new Intent(context, PlanUpdateService.class);
+        intent.putExtra(ARG_LOCAL_BLOG_ID, localTableBlogId);
+        context.startService(intent);
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        AppLog.i(AppLog.T.PLANS, "plan update service > created");
+    }
+
+    @Override
+    public void onDestroy() {
+        AppLog.i(AppLog.T.PLANS, "plan update service > destroyed");
+        super.onDestroy();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        mLocalBlogId = intent.getIntExtra(ARG_LOCAL_BLOG_ID, 0);
+        downloadGlobalPlans();
```

Comment:
Features and Globals Plans are not strictly tied together. We can download them in parallel, no need to serialize them. I think we can use a more aggressive download policy here, and enqueue Features downloading here instead at line 91.